OpenRA.Mods.D2k/Traits/Sandworm.cs

```diff
@@ -91,42 +92,30 @@ void RescanForTargets(Actor self)
 		{
 			targetCountdown = Info.TargetRescanInterval;
 
-			var actorsInRange = self.World.FindActorsInCircle(self.CenterPosition, Info.MaxSearchRadius);
-			var noiseDirection = WVec.Zero;
-
-			foreach (var actor in actorsInRange)
+			// If close enough, we don't care about other actors.
+			var target = self.World.FindActorsInCircle(self.CenterPosition, Info.IgnoreNoiseAttackRange).FirstOrDefault(x => x.HasTrait<AttractsWorms>());
+			if (target != null)
 			{
-				if (!actor.IsInWorld)
-					continue;
-
-				// TODO: Test if we really want to ignore actors that are on rock
-				if (!mobile.Value.CanEnterCell(actor.Location, null, false))
-					continue;
-
-				var noise = actor.TraitOrDefault<AttractsWorms>();
-				if (noise == null)
-					continue;
+				self.CancelActivity();
+				attackTrait.Value.ResolveOrder(self, new Order("Attack", target, true) { TargetActor = target });
+				return;
+			}
 
-				var distance = actor.CenterPosition - self.CenterPosition;
-				var length = distance.Length;
+			Func<Actor, bool> isValidTarget = a =>
+			{
+				if (!a.IsInWorld || a.IsDead)
```

Comment:
Is there a reason why you check this explicit here? (As you don't [check](https://github.com/OpenRA/OpenRA/pull/8090/files#diff-f1cb1831fd3bf409f0aef34c8c5cee8aR96) above.)
Is `FindActorsInCircle` even able to return dead or from the world removed actors?